examples/experimental/i3s-17-and-debug/mesh-layer/simple-mesh-layer-vertex.glsl.js

```diff
@@ -38,10 +38,10 @@ void main(void) {
   geometry.worldPosition = instancePositions;
   geometry.uv = uv;
 
-  #ifdef INSTANCE_PICKING_MODE
-    geometry.pickingColor = instancePickingColors;
+  #ifdef SEGMENTATION_MODE
```

Comment:
I've started thinking about improving the picking system so that we can pick more than one thing at a time. So maybe drop the #ifdefs after all and just add a uniform for now.

Reply:
In this case there are some points which I don't understand:
**- How picking colors can be defined as uniforms if they can be changed per each vertex?**
As I understood uniform it is like a constant which is not changed across drawing. Isn't it?
**- How can I calculate picking colors using uniforms?**
 For example now I am using `attributeManager` with `calculateSegmentaionPickingColors` function inside which creates picking color for each vertex. How should it be implemented in uniforms way?
Can you describe your idea more detailed please?

Reply:
You cannot define picking colors as uniforms, however you can use new a "boolean" uniform to choose which pickingColorArray to use
```suggestion
if (u_pickSegmentation) {
   geometry.pickingColor = instancePickingColors;
} else {
   geometry.pickingColor = segmentationPickingColors;
}
```